src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_LockStatement.cs

```diff
@@ -37,6 +40,41 @@ public override BoundNode VisitLockStatement(BoundLockStatement node)
                     argumentType); //need to have a non-null type here for TempHelpers.StoreToTemp.
             }
 
+            if (argumentType.IsWellKnownTypeLock() &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__EnterLockScope, out MethodSymbol enterLockScope) &&
+                TryGetWellKnownTypeMember(lockSyntax, WellKnownMember.System_Threading_Lock__Scope__Dispose, out MethodSymbol lockScopeDispose))
```

Comment:
Similar comment as for LockBinder.cs.
I think there is no guarantee that the members returned by ```TryGetWellKnownTypeMember``` will have any relationship to the ```argumentType```.